### ISSUE TITLE
🚀 Feature: Authjs with Appwrite adapter

### ISSUE THREAD

**Issue by dwissaaj** [appwrite/appwrite#7564]
### 🔖 Feature description

Feature for adapter working with appwrite database like ~~ @auth/supabase-adapter~~

### 🎤 Pitch

In my use case using auth js really help authenticate,cookies and session user in browser. Event appwrite sdk already have it, it will be greate if we have one
Another Example

- For now appwrite auth  need client component while auth js use server route handler, much more secure
- Auto built in redirect if there is no session
- Multi auth provider in single code
- No need make custom page for login, sign up or register. If using Auth js we can use the default page

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by eldadfux** [comment]
Nice suggestion, would be great if someone from the community could pick this one up.

**Comment by Abhishek-90** [comment]
I would like to take it up, as my first contribution! Please assign it to me!

**Comment by dwissaaj** [comment]
> The description seems less informative, could you please elaborate on it? I would like to take it up.

I have edit more informative, you can check it

**Comment by Abhishek-90** [comment]
From what I gather from your description is, we need to shift auth from client side to server side on server correct?

**Comment by dwissaaj** [comment]
@Abhishek-90 Yes it's also true,  i didn't notice this until when user register in appwrite auth it store the session in local storage, rather not a httpcookies or jwt token in database like auth js

**Comment by dwissaaj** [comment]
> Nice suggestion, would be great if someone from the community could pick this one up.

@Abhishek-90 In Appwrite Init they now support SSR auth which long time ago cannot be done with web SDK. Now we can get Auth js work with Appwrite. Love this update @eldadfux

**Comment by adityaoberai** [comment]
Closing this issue since it was fixed in #7648

**Comment by sh1n-afk** [comment]
Hi, I'd like to work on this issue. I plan to start by implementing the adapter for Auth.js to work with Appwrite, following the example of other similar adapters. Please let me know if there are any specific guidelines or considerations I should be aware of. Thanks!

**Comment by Andres-Ventura** [comment]
No assignor? @stnguyen90

**Comment by abhayear** [comment]
I would like to work on this issue. Could you please assign it to me?

**Comment by Abiji-2020** [comment]
@dwissaaj you are asking a Auth.js adapter for the Appwrite? .
Similar to what we have here right? 
https://authjs.dev/reference/core/adapters

**Comment by dwissaaj** [comment]
> [@dwissaaj](https://github.com/dwissaaj) you are asking a Auth.js adapter for the Appwrite? . Similar to what we have here right? https://authjs.dev/reference/core/adapters

maybe, at that time no auth js for appwrite even you create one it still difficult to implement one due to how appwrite build is diffrent, and old time appwrite sdk is client side and difficult to use, now is easier
